2check/app/components/gm/gm.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from '@angular/core';
+import { Character } from '../../models/character'
+import { CharacterService } from '../../services/character.service';
+import { UpdateService } from '../../services/update.service';
+import faker from 'faker';
+import classList from '../../models/classes';
+
+@Component({
+  selector: 'gm-tools-gm',
+  templateUrl: './gm.component.html',
+  styleUrls: ['./gm.component.css']
+})
+export class GmComponent implements OnInit {
+
+  constructor(public charactersService: CharacterService, private updateService: UpdateService) { }
+
+  characters = [];
+
+  ngOnInit(): void {
+    this.getCharacters();
+  }
+
+  getCharacters() {
+    this.charactersService.getAll().subscribe((respond) => this.characters = respond);
+  }
+
+  deleteCharacter(character: Character) {
+    this.charactersService.delete(character)
+      .subscribe((respond) => {
+        this.characters = this.characters.filter(character =>
+          respond._id !== character._id
+        );
+        this.updateService.sendUpdate("Delete");
+      })
+  }
+
+  saveCharacter(character: Character) {
+    this.charactersService.update(character).subscribe(() => { }), (error) => { alert('NOT SAVED') };
+    this.updateService.sendUpdate("Save");
+  }
+
+  createNewCharacter() {
```

Comment:
you have very big methods in generally related to your busines logic, try to split them into smaller but meaningfull ones, 
one big method can be refactoring by using set of small submethods.